include/openthread/srp_server.h

```diff
@@ -163,7 +169,10 @@ otError otSrpServerSetLeaseRange(otInstance *aInstance,
  * @sa otSrpServerHandleServiceUpdateResult
  *
  */
-typedef void (*otSrpServerServiceUpdateHandler)(const otSrpServerHost *aHost, uint32_t aTimeout, void *aContext);
+typedef void (*otSrpServerServiceUpdateHandler)(otSrpServerServiceUpdateId aId,
```

Comment:
```suggestion
typedef otSrpServerServiceUpdateId (*otSrpServerServiceUpdateHandler)(
```
nit: an alternative solution is let `otSrpServerServiceUpdateHandler` return a `Transaction ID`

Reply:
The SRP Server needs to create the ID and pass it to the Advertising Proxy. The return value will not work.